app/src/main/java/com/theoplayer/android/connector/MainActivity.kt

```diff
@@ -179,6 +188,26 @@ class MainActivity : AppCompatActivity() {
             }
         })
 
+        theoplayerView.player.ads.addEventListener(AdsEventTypes.AD_BREAK_BEGIN) {
+            Log.d("UplynkConnectorEvents", "AD_BREAK_BEGIN ")
+        }
+
+        theoplayerView.player.ads.addEventListener(AdsEventTypes.AD_BREAK_END) {
+            Log.d("UplynkConnectorEvents", "AD_BREAK_END ")
+        }
+
+        theoplayerView.player.ads.addEventListener(AdsEventTypes.AD_BEGIN) {
+            Log.d("UplynkConnectorEvents", "AD_BEGIN ")
+        }
+
+        theoplayerView.player.ads.addEventListener(AdsEventTypes.AD_END) {
+            Log.d("UplynkConnectorEvents", "AD_END ")
+        }
+
+        theoplayerView.player.ads.addEventListener(AdsEventTypes.AD_SKIP) {
+            Log.d("UplynkConnectorEvents", "AD_SKIP ")
+        }
+
```

Comment:
I think all of these are already handled by `setupAdListeners` below? I would suggest we remove all of these `player.ads` listeners inside `setupUplynk`.